gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java

```diff
@@ -227,6 +241,62 @@ private List<String> getEntityIdsToFilterFromSourceEDS(
         .collect(Collectors.toUnmodifiableList());
   }
 
+  private int fetchTotal(
+      ExploreRequestContext requestContext, ExploreRequest request, QueryRequest queryRequest) {
+    QueryRequest totalQueryRequest = buildTotalQueryRequest(request, queryRequest);
+    Iterator<ResultSetChunk> totalIterator = executeQuery(requestContext, totalQueryRequest);
+    while (totalIterator.hasNext()) {
+      ResultSetChunk chunk = totalIterator.next();
+      LOG.info("Total received chunk is: {}", chunk);
+      if (chunk.getRowCount() < 1) {
+        break;
+      }
+
+      if (!chunk.hasResultSetMetadata()
+          || chunk.getResultSetMetadata().getColumnMetadataList().size() != 1
+          || !TOTAL_ALIAS_NAME.equals(
+              chunk.getResultSetMetadata().getColumnMetadata(0).getColumnName())) {
+        LOG.warn("Chunk doesn't have result metadata so couldn't process the response.");
+        break;
+      }
+      return chunk.getRow(0).getColumn(0).getInt();
```

Comment:
Don't think, we would ever cross the `int` limits, but, it's better to have it `getLong` just in case we have an entity that has so many rows (perhaps in the future).